lib/Arch/PPC_C.cpp

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (c) 2022-present Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <anvill/Declarations.h>
+#include <glog/logging.h>
+#include <remill/Arch/Arch.h>
+#include <remill/Arch/Name.h>
+
+#include "AllocationState.h"
+#include "Arch.h"
+
+namespace anvill {
+namespace {
+
+static const std::vector<RegisterConstraint> kParamRegConstraints = {
+    // GPRs
+    RegisterConstraint({VariantConstraint("R3", kTypeIntegral, kMaxBit32)}),
```

Comment:
this seems right but would be nice to provide a reference or something that says the max value is 32 bit even though the registers support 64bit width for our particular processor

Reply:
[88a3367](https://github.com/lifting-bits/anvill/pull/340/commits/88a33677d5f7ce6b9a6020c661ff9c600f5cbe23)

I actually can't find anything that talks about this specifically but I think it's sort of self-explanatory as our toolchain conforms to the 32-bit EABI which is standard on a lot of PPC devices with 32-bit GPRs. I've just linked to the toolchain documentation, not sure if that's good enough.